### PR TITLE
Hook iOS FlutterSpellCheckPlugin to Framework

### PR DESCRIPTION
**Pull request by camsim99**
Fixes the following to make the iOS `FlutterSpellCheckPlugin` compatible with the framework:

- Sets `nativeSpellCheckServiceDefined` to true since iOS supports spell check.
- Translates the Dart locale format to the language iOS format when needed.
- Modifies indices of spell check suggestion spans (there was an off-by-one).

Also removes unnecessary line from Android `SpellCheckPlugin` I encountered.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by cyanglaz** [comment]
Tests added/fixed, removing need test label

File: shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPluginTest.mm [hunk 1]
```diff
@@ -161,11 +161,11 @@ - (void)testFindAllSpellCheckSuggestionsForText {
   XCTAssertTrue(capturedResult.count == 2);
   NSDictionary* suggestionsJSON1 = capturedResult.firstObject;
   XCTAssertEqualObjects(suggestionsJSON1[@"startIndex"], @0);
-  XCTAssertEqualObjects(suggestionsJSON1[@"endIndex"], @4);
+  XCTAssertEqualObjects(suggestionsJSON1[@"endIndex"], @5);
```
Comment (jmagman) [inline shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPluginTest.mm:164]: Is this right?
```
This isss misspelled
xxxxx^^^^xx
01234567890
```
Start index is 5, length is 4, end index is 8, not 9.

Comment (camsim99) [reply shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPluginTest.mm:164]: Yeah, the end is exclusive: https://github.com/flutter/flutter/blob/f57fcac717092e9d55d25a63d79505b455719609/packages/flutter/lib/src/services/spell_check.dart#L19

Comment (jmagman) [reply shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPluginTest.mm:164]: https://api.flutter.dev/flutter/dart-ui/TextRange-class.html
> The next index after the characters in this range.

Well that's confusing, as you experienced from the off by one error...  Can you add a comment about that in `toDictionary`?  In Objective-C this is usually expressed with a tuple range of start index + length, that the endIndex is after the range is not intuitive.

Comment (jmagman) [reply shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPluginTest.mm]:
```
```
Test Case '-[FlutterSpellCheckPluginTest testFindAllSpellCheckSuggestionsForText]' started.
/../../flutter/shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPluginTest.mm:164: error: -[FlutterSpellCheckPluginTest testFindAllSpellCheckSuggestionsForText] : ((suggestionsJSON1[@"endIndex"]) equal to (@4)) failed: ("5") is not equal to ("4")
/../../flutter/shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPluginTest.mm:168: error: -[FlutterSpellCheckPluginTest testFindAllSpellCheckSuggestionsForText] : ((suggestionsJSON2[@"endIndex"]) equal to (@9)) failed: ("10") is not equal to ("9")
Test Case '-[FlutterSpellCheckPluginTest testFindAllSpellCheckSuggestionsForText]' failed (0.061 seconds).
```

Comment (camsim99) [reply shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPluginTest.mm]: @cyanglaz I think I did something weird and reverted your test fixes. Can you re-push them?

**Comment by jmagman** [comment]
Unfortunately I think you're hitting something relating to https://github.com/flutter/engine/pull/35843, can you rebase onto top of tree?

**Review by jmagman — APPROVED** [review]
LGTM

**Comment by zanderso** [comment]
From PR triage: It looks like this PR needs to be rebased as @jmagman says.

**Comment by jmagman** [comment]
Now you're hitting https://github.com/flutter/flutter/issues/111193 😞  you'll have to wait until that's fixed and then rebase or merge again

**Pull request merged**